### DISCUSSION TITLE
Changing the visibility of files in nested folders

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#24214] [Feature Requests]
### Describe the Improvement

Currently, when we click on a folder in the file library we can only see it's own contents.

Is it possible to manage the possibility of viewing the contents of nested folders in a folder in the file library (and even in the file/image fields)?

Let me explain: I'd like it to be possible to have an optional toggle (a checkbox, for example) in a library folder.
Once this box is checked, we'd be able to see all the files contained in the nested folders. If the box is not checked, the behavior remains classic. We can't see the contents of nested folders.

This would enable better management of nested folders, but also when we define a particular folder as the display or upload source in a file or image field.

Thank you for taking the time to read this :)

**Comment by br41nslug** [comment]
Interesting suggestion :D Do you mean showing recursively all files in subfolders or only the next level down? I wonder tho might it be more confusing to see files that are not in the folder you are currently looking at, how would one distinguish between files actually in the current folder vs files in a nested folder?

**Comment by MrCasquette** [comment]
Yes exactly.
This is a common functionnality in some app. I think about Lightroom for example.

let me show an example :

![Capture d’écran 2024-12-11 à 19 01 13](https://github.com/user-attachments/assets/4a1cf00e-8766-4320-afd3-92c9d0164128)

Here I have multiple levels of folder. Currently Mr Topic is empty. It's just a parent folder to group others. I'd like to be able to activate a toggle on this particular folder to see recursively nested folders content.

This would be even more useful when you choose a folder for an image field and it contains numerous subfolders.

As for the distinction, if it's just a visual problem, I don't think it's a problem at all, since you can disable this function on the folder with a simple toggle.
We could possibly think of an outline or some classic visual feedback, but that seems like overkill to me.

**Comment by AlexGaillard** [comment]
I'd imagine we could show an icon on the image card, and likely include the toggle in the header without too much difficulty.<br><br>However there is also the "All files" option, though it isn't specific to a folder as you're suggesting.

<img src="https://uploads.linear.app/8d64f771-381d-4a46-a408-28b4fcc13d22/dbff7f8a-67cc-40d8-9783-2514aa5571ef/f0ecd5a1-8ec1-44fe-81bb-6f2b9d3506e8?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzhkNjRmNzcxLTM4MWQtNGE0Ni1hNDA4LTI4YjRmY2MxM2QyMi9kYmZmN2Y4YS02N2NjLTQwZDgtOTc4My0yNTE0YWE1NTcxZWYvZjBlY2Q1YTEtOGVjMS00NGZlLTgxYmItNmYyYjlkMzUwNmU4IiwiaWF0IjoxNzMzOTQ3MTIzLCJleHAiOjMzMzA0NTA3MTIzfQ.d-2BvQZUsCaff8fKjhw9cC0eB5zWYf29Q7c_Vc1xIwE " alt="image.png" width="132" data-linear-height="52" />

**Comment by MrCasquette** [comment]
> I'd imagine we could show an icon on the image card, and likely include the toggle in the header without too much difficulty.<br><br>However there is also the "All files" option, though it isn't specific to a folder as you're suggesting.
> 
> <img src="https://uploads.linear.app/8d64f771-381d-4a46-a408-28b4fcc13d22/dbff7f8a-67cc-40d8-9783-2514aa5571ef/f0ecd5a1-8ec1-44fe-81bb-6f2b9d3506e8?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzhkNjRmNzcxLTM4MWQtNGE0Ni1hNDA4LTI4YjRmY2MxM2QyMi9kYmZmN2Y4YS02N2NjLTQwZDgtOTc4My0yNTE0YWE1NTcxZWYvZjBlY2Q1YTEtOGVjMS00NGZlLTgxYmItNmYyYjlkMzUwNmU4IiwiaWF0IjoxNzMzOTQ3MTIzLCJleHAiOjMzMzA0NTA3MTIzfQ.d-2BvQZUsCaff8fKjhw9cC0eB5zWYf29Q7c_Vc1xIwE " alt="image.png" width="132" data-linear-height="52" />

Very good idea for the icon.
But let's try looking for a file in the "all file" without knowing the name 😅 or if you just want to browse and find the idéal image ^^

**Comment by rijkvanzanten** [comment]
Interesting suggestion. I'll move it to feature requests while we're still discussing what this could look like and how other apps are handling this 🙂
